zephyr/Kconfig

```diff
@@ -3,33 +3,46 @@
 
 if SOC_FAMILY_ESPRESSIF_ESP32
 
-if MCUBOOT
+config EFUSE_VIRTUAL
+	bool "Simulate eFuse operations in RAM"
+	help
+	  If "y" - The virtual mode is enabled and all eFuse operations (read and write) are redirected
+	  to RAM instead of eFuse registers, all permanent changes (via eFuse) are disabled.
+
+	  If it is "y", then SECURE_FLASH_ENCRYPTION_MODE_RELEASE cannot be used.
+	  Because the EFUSE VIRT mode is for testing only.
+
+	  During startup, the eFuses are copied into RAM. This mode is useful for fast tests.
+
+config EFUSE_VIRTUAL_KEEP_IN_FLASH
+	bool "Keep eFuses in flash"
+	depends on EFUSE_VIRTUAL
+	help
+	  In addition to the "Simulate eFuse operations in RAM" option, this option just adds
+	  a feature to keep eFuses after reboots in flash memory. To use this mode the partition_table
+	  should have the `sys_partition` partition.
+
+	  During startup, the eFuses are copied from flash or,
+	  in case if flash is empty, from real eFuse to RAM and then update flash.
+	  This mode is useful when need to keep changes after reboot
+	  (testing secure_boot and flash_encryption).
 
 config EFUSE_VIRTUAL_OFFSET
 	hex
-	default 0x250000 if SOC_SERIES_ESP32
-	default 0x250000 if SOC_SERIES_ESP32S2
-	default 0x250000 if SOC_SERIES_ESP32S3
-	default 0x250000 if SOC_SERIES_ESP32C3
-	default 0x250000 if SOC_SERIES_ESP32C6
+	depends on EFUSE_VIRTUAL_KEEP_IN_FLASH
+	default $(dt_nodelabel_reg_addr_hex,sys_partition) if $(dt_nodelabel_exists,sys_partition)
+	default 0x0
 
 config EFUSE_VIRTUAL_SIZE
 	hex
-	default 0x2000 if SOC_SERIES_ESP32
-	default 0x2000 if SOC_SERIES_ESP32S2
-	default 0x2000 if SOC_SERIES_ESP32S3
-	default 0x2000 if SOC_SERIES_ESP32C3
-	default 0x2000 if SOC_SERIES_ESP32C6
+	depends on EFUSE_VIRTUAL_KEEP_IN_FLASH
```

Comment:
I think the size of the eFuses should stay fixed and not depend on the size of sys_partition.